### PR TITLE
Consolidate and config a rendering GHA 

### PR DESCRIPTION
**Pull request by cansavvy**
#### What changes are being implemented in this Pull Request?
This rendering yaml builds upon the idea in #445 and makes another BIG pull request yaml that reads in the config_automation.yml to determine whether checks should be run or not. 

Now pull-request.yml contains all the GHA that run upon opening merging to main. This includes 43GHA:
- `render-bookdown` 
- `render-coursera` - this is dependent on bookdown running
- `render-leanpub` - this is dependent on coursera running 

Also in this PR: 
- Updates to sync.yml 
- Delete all `TRIGGER-START/END`'s

### CODE REVIEW

File: .github/workflows/delete-preview.yml [hunk 1]
```diff
@@ -19,8 +17,7 @@ jobs:
         shell: bash
 
       # Delete the branch!
-      - name: Delete branch
-        uses: dawidd6/action-delete-branch@v3
-        with:
-          github_token: ${{ secrets.GH_PAT }}
-          branches: preview-${{ github.event.pull_request.number }}
+      - name: Delete branch locally and remotely
```
Comment (cansavvy) [inline .github/workflows/delete-preview.yml:20]: I noticed that the canned github action `dawidd6/action-delete-branch@v3` has been archived, so we probably should move away from using that and just this the old fashioned way.

File: .github/workflows/docker-build.yml [hunk 2]
```diff
@@ -1,23 +1,14 @@
 # Candace Savonen Apr 2021
 
-name: Build of Docker
+name: Build Docker
```
Comment (cansavvy) [inline .github/workflows/docker-build.yml]: This on its own is still handy to have to run outside of `pull-request.yml` (I use it often to rebuild images and push them to dockerhub). But it was able to be simplified now that it doesn't need to run on pull request. (the version that runs on pull request is in the pull-request.yml See #448)

**Comment by cansavvy** [comment]
It unfortunately looks like I won't be able to truly test this until its merged to main, but still worth reviewing it beforehand.

**Comment by avahoffman** [comment]
I'll test this asap! Thanks for making the ideas a reality! 🤩

File: .github/workflows/render-all.yml [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+      - name: Load environment from YAML
+        uses: doughepi/yaml-env-action@v1.0.0
+        with:
+            files: config_automation.yml # Pass a space-separated list of configuration files. Rightmost files take precedence.
+    outputs:
+      toggle_bookdown: "${{ env.RENDER_BOOKDOWN }}"
+      toggle_coursera: "${{ env.RENDER_COURSERA }}"
+      toggle_leanpub: "${{ env.RENDER_LEANPUB }}"
+
+  render-bookdown:
```
Comment (avahoffman) [inline .github/workflows/render-all.yml:33]:
```suggestion
  render-bookdown:
    name: Render bookdown
```

File: .github/workflows/render-all.yml [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+          exit 1
+
+      # Commit the rendered bookdown files
+      - name: Commit rendered bookdown files
+        run: |
+          git config --local user.email "actions@github.com"
+          git config --local user.name "GitHub Actions"
+          git add --force docs/*
+          git commit -m 'Render bookdown' || echo "No changes to commit"
+          git push origin main || echo "No changes to push"
```
Comment (avahoffman) [inline .github/workflows/render-all.yml:79]: I'm assuming this is just copied over from the old yamls? Did you make any big changes?

File: .github/workflows/render-all.yml [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+  render-coursera:
```
Comment (avahoffman) [inline .github/workflows/render-all.yml:81]:
```suggestion
  render-coursera:
    name: Render Coursera
```

File: .github/workflows/render-all.yml [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+      # Commit the rendered bookdown files
+      - name: Commit rendered Coursera files
+        run: |
+          git config --local user.email "actions@github.com"
+          git config --local user.name "GitHub Actions"
+          git add --force docs/coursera/*
+          git commit -m 'Render Coursera' || echo "No changes to commit"
+          git push origin main || echo "No changes to push"
+
+  render-leanpub:
```
Comment (avahoffman) [inline .github/workflows/render-all.yml:123]:
```suggestion
  render-leanpub:
    name: Render Leanpub
```

**Review by avahoffman — APPROVED** [review]
Crushed it!! 💪🏻 LGTM!

File: .github/workflows/render-all.yml (hunk 4 above)
Comment (cansavvy) [reply .github/workflows/render-all.yml:79]: Just copied over. GitHub can't track that apparently.

Comment (avahoffman) [reply .github/workflows/render-all.yml:79]: Yeah I wish there was a better way to track snippets moved from file to file. Get with the program github 😛

**Pull request merged**